com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/liquibase/PopulateGUIDColumnConfigurationRegistryChange.java

```diff
@@ -0,0 +1,125 @@
+package com.sap.cloud.lm.sl.cf.core.liquibase;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.cloudfoundry.client.lib.CloudCredentials;
+import org.cloudfoundry.client.lib.domain.CloudSpace;
+import org.cloudfoundry.client.lib.rest.CloudControllerClient;
+import org.cloudfoundry.client.lib.rest.CloudControllerClientFactory;
+
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.model.CloudTarget;
+import com.sap.cloud.lm.sl.cf.core.util.Configuration;
+import com.sap.cloud.lm.sl.cf.core.util.SecurityUtil;
+
+public class PopulateGUIDColumnConfigurationRegistryChange
+    extends AbstractDataTransformationChange<Map<Long, CloudTarget>, Map<Long, String>> {
+
+    private static final String SELECT_STATEMENT = "SELECT ID, TARGET_SPACE, TARGET_ORG FROM CONFIGURATION_REGISTRY LIMIT 100";
+    private static final String UPDATE_STATEMENT = "UPDATE CONFIGURATION_REGISTRY SET SPACE_ID=? WHERE ID=?";
+
+    @Override
+    public Map<Long, String> transformData(Map<Long, CloudTarget> retrievedData) {
+
+        HashMap<Long, String> result = new HashMap<>();
+        CloudControllerClient cfClient = getCFClient();
+        List<CloudSpace> spaces = getSpaces(cfClient);
+
+        String org, space;
+        for (Long id : retrievedData.keySet()) {
+            org = retrievedData.get(id).getOrg();
+            space = retrievedData.get(id).getSpace();
+            if (StringUtils.isEmpty(space)) {
+                continue;
+            }
+
+            if (StringUtils.isEmpty(org)) {
+                result.put(id, space);
+            } else {
+                String spaceGuid = getSpaceGuid(org, space, spaces);
+                result.put(id, spaceGuid);
+            }
+            
+        }
+        return result;
+    }
+
+    protected List<CloudSpace> getSpaces(CloudControllerClient cfClient) {

```

Comment:
This is probably used for testing purposes right? If you use CloudFoundryClientFactory, you can mock it to return a mocked CF client and this method would then be unnecessary.